src/libstore/filetransfer.hh

```diff
@@ -56,6 +56,8 @@ struct FileTransferRequest
     std::string expectedETag;
     bool verifyTLS = true;
     bool head = false;
+    std::string authmethod;
```

Comment:
This should be an `enum struct`.

Reply:
Uploaded my attempt at this, however I don't write much C/C++ code.

I tried several different ways to make this a `Setting<HttpAuthMethod>` using the templates for `BaseSetting`, but I could not get this to work for a URL type parameters, and indeed none of the other url parameters are using enums.